homeassistant/components/mqtt/cover.py

```diff
@@ -347,26 +339,57 @@ def position_message_received(msg):
                 template = self._config.get(CONF_VALUE_TEMPLATE)
 
             if template is not None:
-                payload = template.async_render_with_possible_json_value(payload)
-
-            if not payload:
-                _LOGGER.debug("Ignoring empty position message from '%s'", msg.topic)
-                return
+                variables = {
+                    "entity_id": self.entity_id,
+                    "position_open": self._config[CONF_POSITION_OPEN],
+                    "position_closed": self._config[CONF_POSITION_CLOSED],
+                    "tilt_min": self._config[CONF_TILT_MIN],
+                    "tilt_max": self._config[CONF_TILT_MAX],
+                }
+                payload = template.async_render_with_possible_json_value(
+                    payload, variables=variables
+                )
 
-            if payload.isnumeric():
+                if not payload:
+                    _LOGGER.debug(
+                        "Ignoring empty position message from '%s'", msg.topic
+                    )
+                    return
+
+                try:
+                    payload = json_loads(payload)
+                except (TypeError, JSONDecodeError):
```

Comment:
```suggestion
                except JSONDecodeError:
```
template should return a string so `TypeError` should not happen